mender-update/context/context.cpp

```diff
@@ -37,6 +37,16 @@ namespace json = mender::common::json;
 namespace kv_db = mender::common::key_value_database;
 namespace path = mender::common::path;
 
+const string MenderContext::artifact_name_key {"artifact-name"};
```

Comment:
Yeah, same question here. I even believe I tried this and there was some problem with it. Did you try adding them as `extern` to the `.hpp` file?

Reply:
To avoid allocating a new instance every time `MenderContext` is initialized. They are all const, so we might as well allocate them once and be done with it. `extern` would probably work, but it's redundant in this case (it is automatically assumed for static class members).

If you're asking about the initializer in the hpp file, then the reason I had to move them is that they don't work for static members. Keeping them would of course be nice for readability, but it's a language limitation.

Reply:
I see makes sense. However, in the grand scheme of things, what difference does this optimization really make?

Reply:
Not much, since `MenderContext` is typically only constructed once. However, since they are now globals you don't need a `MenderContext` instance at all to refer to them. I'm using this in several places, for example `MenderContext::standalone_state_key`.

Reply:
> If you're asking about the initializer in the hpp file, then the reason I had to move them is that they don't work for static members. Keeping them would of course be nice for readability, but it's a language limitation.

Right, that's the thing I was hitting and how I ended up putting them into the class as normal members.